core/dbt/utils.py

```diff
@@ -386,3 +386,21 @@ def strtobool(val: str) -> bool:
         return False
     else:
         raise ValueError("invalid truth value %r" % (val,))
+
+
+def try_get_max_rss_kb() -> Optional[int]:
+    """Attempts to get the high water mark for this process's memory use via
+    the most reliable and accurate mechanism available through the host OS.
+    Currently only implemented for Linux."""
+    try:
```

Comment:
Maybe check for `sys.platform == 'linux'` in the first line to avoid the `try/catch`?